docs/docs/how_to/how-to-oracles.md

```diff
@@ -0,0 +1,270 @@
+---
+title: How to use Oracles
+description: Learn how to use oracles in your Noir program with examples in both Nargo and NoirJS. This guide also covers writing a JSON RPC server and providing custom foreign call handlers for NoirJS.
+keywords:
+  - Noir Programming
+  - Oracles
+  - Nargo
+  - NoirJS
+  - JSON RPC Server
+  - Foreign Call Handlers
+sidebar_position: 1
+---
+
+This guide shows you how to use oracles in your Noir program. For the sake of clarity, it is assumed that:
+
+- You have read the [explainer on Oracles](../explainers/explainer-oracle.md) and are comfortable with the concept
+- You already have your own NoirJS and/or Nargo app. If you don't, you can use the [vite-hardhat starter](https://github.com/noir-lang/noir-starter/tree/main/vite-hardhat) as a boilerplate (for example, by using its [devcontainer in codespaces](./using-devcontainers.mdx))
+- You understand the concept of a JSON-RPC server. Visit the [JSON-RPC website](https://www.jsonrpc.org/) if you need a refresher.
+- You are comfortable with server-side javascript. Will skip any details on installing Node, packages, etc, so as to keep the guide short and straight to the point.
+
+For reference, you can find the snippets used in this tutorial on the [Aztec DevRel Repository](https://github.com/AztecProtocol/dev-rel/tree/main/how_to_oracles/code-snippets/how-to-oracles)
+
+## Rundown
+
+This guide has 3 major steps:
+
+1. How to modify our circuit to make use of oracle calls as unconstrained functions.
+2. How to write a JSON RPC Server to resolve these oracle calls with Nargo
+3. How to use them in Nargo and how to provide a custom resolver in NoirJS
+
+## Step 1 - Modify your circuit
+
+An oracle is defined in a circuit by defining two methods:
+
+- An unconstrained method - This tells the Noir compiler that it is executing a Brillig block.
+- A decorated oracle method - This tells the compiler that this method is an RPC call.
+
+An example of an oracle that returns a `Field` would be:
+
+```rust
+#[oracle(getSquared)]
+unconstrained fn squared(number: Field) -> Field { }
+
+unconstrained fn get_squared(number: Field) -> Field {
+    squared(number)
+}
+```
+
+In this example, we're wrapping our oracle function in a unconstrained method, and decorating it with `oracle(getSquared)`. We can then call the unconstrained function as we would call any other function:
+
+```rust
+fn main(input: Field) {
+    let squared = get_squared(input);
+}
+```
+
+In the next section, we will make this `getSquared` be a method of the RPC server Noir will use.
```

Comment:
It'd be useful to call out that the RPC server will use the method defined in the decorator, not the name of the Noir function. This confused me for more than a minute.

```suggestion
In the next section, we will make this `getSquared` (defined on the `squared` decorator) be a method of the RPC server Noir will use.